packages/components/src/internal/components/user/UsersGridPanel.tsx

```diff
@@ -173,20 +173,24 @@ export class UsersGridPanelImpl extends PureComponent<Props, State> {
         this.closeDialog();
         this.onRowSelectionChange(this.getUsersModel(), undefined, false); // clear selected user details
         this.props.onCreateComplete(response, roles);
-        this.reloadUsersModel();
+        this.props.actions.onModelChange(this.getUsersModelId(), { changeType: ChangeType.add });
     };
 
     reloadUsersModel(): void {
         this.props.actions.loadModel(this.getUsersModelId(), true, true);
     }
 
-    onUsersStateChangeComplete = (response: any, resetSelection = true): void => {
+    onUsersStateChangeComplete = (response: any, isDelete: boolean = false): void => {
         this.closeDialog();
-        if (resetSelection) {
-            this.onRowSelectionChange(this.getUsersModel(), undefined, false); // clear selected user details
-        }
+        this.updateSelectedUserId(undefined); // clear selected user details
         this.props.onUsersStateChangeComplete(response);
-        this.reloadUsersModel();
```

Comment:
is `reloadUsersModel` still used in this component after these usages were changed or can it be removed?

Reply:
it's still used in componentDidUpdate, and I believe that usage is correct.